### ISSUE TITLE
Unable to generate multipoles using Python API

### ISSUE THREAD

**Issue by VHchavez** [psi4/psi4#2706]
Hello folks, 

I am trying to obtain some multipoles using `psi4.gdma`. While using psi4 `1.6.1` I am able to run [this](https://github.com/psi4/psi4/blob/master/samples/gdma/gdma1/input.dat) example. 
Unfortunately, the program unexpectedly fails when running the **Python** snippet shown below, resulting in no information. 

Any advice on how to troubleshoot this issue?
Thanks :)

Here's what I'm running:

```
import psi4
#psi4.core.be_quiet()

water = psi4.geometry("""
    O  0.000000  0.000000  0.117176
    H -0.000000 -0.756950 -0.468706
    H -0.000000  0.756950 -0.468706
noreorient
nocom
""")

psi4.set_options({"scf_type"      : "pk",
                  "basis"         : "cc-pvtz",
                  "d_convergence" : 10,
})

psi4.set_options({ "gdma_switch" : 0,
                   "gdma_radius" : [ "H", 0.65 ],
                   "gdma_limit"  : 2,
                   "gdma_origin" : [0.0, 0.0, 0.117176]

})


e, wfn = psi4.energy("scf", return_wfn=True, molecule=water)
psi4.gdma(wfn)

print("I did not die while attempting to generate multipoles")
```

**Comment by JonathonMisiewicz** [comment]
I need to see _exactly_ what this "analogous snippet in Python" is.

**Comment by loriab** [comment]
I think the link in OP was the Psithon and the snippet in OP was the "analogous Python". 

I can reproduce your findings when running as `python psiapi.py`. It _does_ seem to work fine when run as `psi4 psiapi.py`. Still puzzling that out.

```
> stage/bin/psi4 tmpgdma.py 
AA
BB
I did not die while attempting to generate multipoles
[[ 6.06050501e-09 -7.96653157e-01  5.23637240e-16  5.11358565e-16
   1.08132595e-01 -5.88257502e-16 -1.20499132e-15 -2.01989584e+00
  -6.92266294e-17]]
```

**Comment by VHchavez** [comment]
@JonathonMisiewicz, @loriab is correct. The "analogous snippet" is the code shown at the end of the post. I apologize for my phrasing. 
I attempted to run in another env with psi4 `1.3.2` with the same issue.

**Comment by zachglick** [comment]
I think I have a partial understanding of why the python snippet works with the psi4 executable, but not psi4's python api. I suspect something about psi4's GDMA interface requires/assumes that there is an output file. When you run psi4 as an executable, the results are written to an output file, so everything is fine. When you run psi4 through the python api, the results are by default printed to your terminal and there is no output file, so you run into this problem.

Here's a workaround. Add the following line to the python snippet, immediately after the `import psi4` line:
```
psi4.core.set_output_file('output.dat', False)
```
This line changes the default output of psi4's python api (from the terminal to a file "output.dat"). When I insert this line, the python snippet runs successfully.

Someone more familiar with the GDMA interface than myself should look into how to fix psi4 so that this workaround isn't necessary.

**Comment by VHchavez** [comment]
Amazing, this works. Thanks Zack ☺️!!

**Comment by loriab** [comment]
Wow, looks like @hokru solved this 4 years ago in #1334. I've merged the PR to psi4/gdma but will need to rebuild the stack before you see any improvement.

> Is there any documentation on how to keep an isolated installation of psi4 for development? Any tips would be appreciated!

It doesn't look like you need to do any investigation for this problem, but I'm glad to help on this in general. I'm not sure what you mean by "isolated installation of psi4 for development". I manage everything with conda envs. One might have an env with a psi4 conda package installed to use at runtime and another env with psi4-dev conda package installed to supply deps for a psi4 source repository with a compile directory.

**Comment by VHchavez** [comment]
Closing since it was been fixed and merged. 
Thank you for your help :)
